library/psa_crypto.c

```diff
@@ -93,7 +93,102 @@ static inline int safer_memcmp( const uint8_t *a, const uint8_t *b, size_t n )
     return( diff );
 }
 
+#if defined(MBEDTLS_PSA_BUILTIN_ALG_AES_DRBG)
+/*
+ * Map the PSA definitions to the mbedtls CTR DRBG versions if the
+ * built in version of the AES DRBG feature is selected
+ */
+typedef struct mbedtls_ctr_drbg_context psa_ctr_drbg_context;
+
+#define psa_ctr_drbg_init mbedtls_ctr_drbg_init
+#define psa_ctr_drbg_free mbedtls_ctr_drbg_free
+#define psa_ctr_drbg_seed mbedtls_ctr_drbg_seed
+#define psa_ctr_drbg_random mbedtls_ctr_drbg_random
+
+#else
+
+/**
+ * \brief          The PSA_CTR_DRBG context structure.
+ */
+typedef struct psa_ctr_drbg_context
+{
+    unsigned char counter[16];  /*!< The counter (V). */
+    int reseed_counter;         /*!< The reseed counter.
+                                 * This is the number of requests that have
+                                 * been made since the last (re)seeding,
+                                 * minus one.
+                                 * Before the initial seeding, this field
+                                 * contains the amount of entropy in bytes
+                                 * to use as a nonce for the initial seeding,
+                                 * or -1 if no nonce length has been explicitly
+                                 * set (see mbedtls_ctr_drbg_set_nonce_len()).
+                                 */
+    int prediction_resistance;  /*!< This determines whether prediction
+                                     resistance is enabled, that is
+                                     whether to systematically reseed before
+                                     each random generation. */
+    size_t entropy_len;         /*!< The amount of entropy grabbed on each
+                                     seed or reseed operation, in bytes. */
+    int reseed_interval;        /*!< The reseed interval.
+                                 * This is the maximum number of requests
+                                 * that can be made between reseedings. */
+
+    mbedtls_aes_context aes_ctx;        /*!< The AES context. */
+
+    /*
+     * Callbacks (Entropy)
+     */
+    int (*f_entropy)(void *, unsigned char *, size_t);
+                                /*!< The entropy callback function. */
+
+    void *p_entropy;            /*!< The context for the entropy function. */
+
+#if defined(MBEDTLS_THREADING_C)
+    mbedtls_threading_mutex_t mutex;
+#endif
+}
+psa_ctr_drbg_context;
+
+/*
+ * TBD: These functions need to be filled out with the support required to
+ * use driver acceleration for DRBG support.
```

Comment:
Yes, they do. I'm confused: is the PR ready or not? I don't see how the PR is useful without implementing these functions.